js/reducers/index.js

```diff
@@ -174,14 +172,19 @@ function report(state = INITIAL_REPORT_STATE, action) {
       return state.set("answers", Immutable.fromJS(preprocessAnswersJSON(action.response)));
     case SET_NOW_SHOWING:
       return state.set("nowShowing", action.value);
-    case SET_QUESTION_SELECTED:
-      return state.setIn(["questions", action.key, "selected"], action.value);
+
+    // The following actions only trigger API middleware hooks that call out
+    // to firebase. The results of the calls are handled by RECEIVE_USER_SETTINGS
+    case SET_ANONYMOUS:
     case HIDE_UNSELECTED_QUESTIONS:
-      return hideUnselectedQuestions(state);
     case SHOW_UNSELECTED_QUESTIONS:
-      return showUnselectedQuestions(state);
-    case SET_ANONYMOUS:
-      return setAnonymous(state, action.value);
+    case SET_QUESTION_SELECTED:
+      return state; // These ⬆ API-calling actions just return the current state.
```

Comment:
I think we don't need to list these actions here, as they don't do anything. `default` handler simply returns state.
But I don't understand how that works now - does it mean that once a user selects a question, it will be locally selected only when we get a response from Firestore (using RECEIVE_USER_SETTINGS handler)? Then it'd depend on network conditions, so I'm not sure about that.

Reply:
Because the `RECEIVE_USER_SETTINGS` call is a side-effect of the API middleware invocation, I wanted to put documentation of that in the actions handler. 

This approach works when detached from the network.  `RECEIVE_USER_SETTINGS` is called with the data we expect. as if we were able to connect to FireStore:

> Local writes in your app will invoke snapshot listeners immediately. This is because of an important feature called "latency compensation." When you perform a write, your listeners will be notified with the new data before the data is sent to the backend.  [link](https://firebase.google.com/docs/firestore/query-data/listen)

Reply:
Okay, that makes sense. And it's nice, since it can simplify initial settings initialization and update process, as everything is handled just by one `RECEIVE_USER_SETTINGS` handler.
Maybe I'd add some comment somewhere that this Firestore update resolves immediately, so it's clear.